### PR TITLE
[Wallet] Make invite redemption more robust

### LINKED ISSUE #884: Improve reliability of redeem invite
### Expected Behavior

Fewer failures, better error handling

### Current Behavior

Even after recent improvements, redeem invites still sometimes fail and leave the code unusable.

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Improve error and edge case handling of redeem invite
- Speed up invite redemption a little bit
- Improve code organization or redemption saga
- Minor style tweaks to EnterInviteCode.tsx
- Remove dead code from JoinCelo.tsx
- Tweaks navigateToUri's function signature
- Move stripHexLeader utility in utils

### Tested

Redeemed invite code successfully
Attempted to use empty invite code
Loaded invite screen after invite already redeemed

### Related issues

- Fixes #884 

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/.env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ENVIRONMENT=local
-DEFAULT_TESTNET=alfajoresstaging
+DEFAULT_TESTNET=pilotstaging
```
Comment (jeanregisser) [inline packages/mobile/.env]: Should this be part of this PR?

File: packages/mobile/src/invite/JoinCelo.tsx [hunk 2]
```diff
@@ -214,8 +187,8 @@ export class JoinCelo extends React.Component<Props, State> {
           standard={false}
           type={BtnTypes.PRIMARY}
           text={t('continue')}
-          onPress={this.submit}
-          disabled={this.state.isSubmitting || !this.state.isValidNumber}
+          onPress={this.onPressContinue}
+          disabled={!this.state.isValidNumber}
```
Comment (jeanregisser) [inline packages/mobile/src/invite/JoinCelo.tsx:191]: Wondering what happens if you try to press it quickly? Does it go twice to the next screen?

File: packages/mobile/src/utils/linking.ts [hunk 3]
```diff
@@ -3,30 +3,27 @@ import DeviceInfo from 'react-native-device-info'
...
 
 export function navigateToWalletPlayStorePage() {
   Linking.openURL(`market://details?id=${DeviceInfo.getBundleId()}`)
 }
 
-interface NavigateToURIInterface {
-  uri: string
-  backupUri?: string
-}
+export async function navigateToURI(uri: string, backupUri?: string) {
```
Comment (jeanregisser) [inline packages/mobile/src/utils/linking.ts:16]: When reading `navigateToURI('http://someplace', 'http://someotherplace')` it's not exactly clear what the second param does. 

May I suggest `navigateToURI('http://someplace', { backupUri: 'http://someotherplace' })` instead?

File: packages/utils/src/async.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export const retryAsync = async (
   delay = 100
 ) => {
   let saveError
-  for (let i = 0; i < tries + 1; i++) {
+  for (let i = 0; i < tries; i++) {
```
Comment (jeanregisser) [inline packages/utils/src/async.ts:19]: Nice catch! 👍

**Review by jeanregisser — APPROVED** [review]
Nice work! 🎉

File: packages/mobile/src/invite/JoinCelo.tsx (hunk 2 above)
Comment (jmrossy) [reply packages/mobile/src/invite/JoinCelo.tsx:191]: tested, seems fine.

**Pull request merged**